packages/make-styles/src/runtime/utils/hashClassName.ts

```diff
@@ -0,0 +1,24 @@
+import { hashString } from './hashString';
+import { HASH_PREFIX } from '../../constants';
+
+export interface HashedClassNameParts {
+  property: string;
+  value: string;
+  pseudo: string;
+  media: string;
+  support: string;
+  unstable_cssPriority: number;
+}
+
+export function hashClassName({
+  media,
+  property,
+  pseudo,
+  support,
+  value,
+  unstable_cssPriority,
+}: HashedClassNameParts): string {
```

Comment:
IMO we should focus a bit more in readability, otherwise maintenability will be compromised.

Can't we rely on uglyfiers to simply minimize even internal properties of objects?! I think it would be a good solution for this, since the external API is quite clear.

The way I see this code will become more and more complex to mantain if we don't start worrying a little bit more about readability.